expr/functions/moving/function.go

```diff
@@ -144,22 +160,20 @@ func (f *moving) Do(ctx context.Context, e parser.Expr, from, until int64, value
 		cons = "min"
 	case "movingMax":
 		cons = "max"
-	}
-
-	if len(arg) == 0 {
-		return nil, nil
+	case "movingMedian":
+		cons = "median"
 	}
 
 	var offset int
 
-	if scaleByStep {
+	if e.Arg(1).Type() == parser.EtString {
 		windowSize /= int(arg[0].StepTime)
-		offset = windowSize
 	}
+	offset = windowSize
 
 	result := make([]*types.MetricData, len(arg))
```

Comment:
Should we update this to.. ?

```suggestion
	result := make([]*types.MetricData, len(adjustedArgs))
```

I'm not sure if they can be different, but just in case (I'm thinking they can, if a series stops existing just in that gap between the start and the adjustedStart)